low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -20,85 +20,293 @@
+from framework.base.module_thread import SensorThread, ThreadException
 from framework.base.sspl_constants import DATA_PATH
-from framework.utils.conf_utils import ( GLOBAL_CONF, SSPL_CONF,
-    Conf, NODE_ID_KEY)
-from framework.utils.severity_reader import SeverityReader
-from framework.utils.mon_utils import get_alert_id
+from framework.utils.conf_utils import (SSPL_CONF, Conf)
 from framework.utils.iem import Iem
-from framework.base.module_thread import ThreadException
-from framework.utils.store_factory import file_store
+from framework.utils.mon_utils import get_alert_id
+from framework.utils.service_logging import logger
+from framework.utils.severity_reader import SeverityReader
+from framework.utils.store_factory import store
+from message_handlers.service_msg_handler import ServiceMsgHandler
+from sensors.ISystem_monitor import ISystemMonitor
+
+UNIT_IFACE = "org.freedesktop.systemd1.Unit"
+SERVICE_IFACE = "org.freedesktop.systemd1.Service"
+MANAGER_IFACE = 'org.freedesktop.systemd1.Manager'
+SYSTEMD_BUS = "org.freedesktop.systemd1"
+CACHE_PATH = f"/{DATA_PATH}/server/service_monitor/"
+
+Alert = namedtuple('Alert', 'alert_type description impact recommendation')
+FailedAlert = Alert("fault", "{} in {} state.", "{} service is unavailable.",
+                    "Try to restart the service")
+InactiveAlert = Alert("fault", "{} in {} state for more than {} seconds.",
+                      "{} service is unavailable.",
+                      "Try to restart the service")
+ResolvedAlert = Alert("fault_resolved", "{} in {} state.",
+                      "{} service is available now.",
+                      "No action required")
+
+
+class ActiveState:
+    """
+    State class representing monitoring state Active and actions needed to do
+    when service becomes active.
+    """
+
+    @staticmethod
+    def enter(service):
+        Service.non_active.discard(service.name)
+
+
+class FailedState:
+    """
+    State class representing monitoring state Failed and actions needed to do
+    when service raises alert(nonactive for threshold time or failed directly).
+    """
+
+    @staticmethod
+    def enter(service):
+        Service.non_active.discard(service.name)
+
+
+class InactiveState:
+    """
+    State class representing monitoring state Inactive and actions needed to do
+    when service change state other than active or failed.
+    """
+
+    @staticmethod
+    def enter(service):
+        service.nonactive_enter_timestamp = time.time()
+        Service.non_active.add(service.name)
+
+
+class DisabledState:
+    """
+    State class representing monitoring state Disabled and actions needed to do
+    when service is disabled.
+    When service enters into Disabled state, it removes signal. So, monitoring
+    for particular service will be stopped.
+    """
+
+    @staticmethod
+    def enter(service):
+        logger.warning("{} service is disabled, it will not be "
+                       "monitored".format(service.name))
```

Comment:
Do we need to send an INFO alert if service is enabled/disabled at run-time??

Reply:
Aren't we raising alert on state transition from enabled -> disabled or vice versa. Yes we need to

Reply:
Why we need to raise alert for enabled-disabled ? It has nothing to do with actual service state. How it is going to be useful ?

Reply:
@sandeepsgt  disabling service alters its behavior on reboot, which may need to be notified. Same applies to enabled state

Reply:
Though its not mentioned in spec, so  we can discuss over this first and take it up as future improvement if needed, keep a TODO to consider in code itself

Reply:
Requirement not in current spec. Added todo to revisit in future